### PR TITLE
feat: add setNDKCrashesEnabled API

### PR DESCRIPTION
**Pull request by ahmedAlaaInstabug**
## Description of the change

1) Map setNDKCrashesEnabled in Javascript side

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Jira ID:MOB-13774

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

[MOB-13774]: https://instabug.atlassian.net/browse/MOB-13774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mnoseir** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  You have not included a CHANGELOG entry! 
You can find it at [CHANGELOG.md](https://github.com/Instabug/Instabug-React-Native/blob/master/CHANGELOG.md).
  </td>
    </tr>
  </tbody>
</table>


### Coverage Report
| Label | Coverage | Status |
| ----- | -------- | ------ |
| JavaScript | 96.3% | ✅ |
| Android | 47.5% | ✅ |
| iOS | 41.3% | ✅ |
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f73bd74598d41905a7084e43866d57fdda63b92d
</p>

**Review by abdelhamid-f-nasser — CHANGES_REQUESTED** [review]
Great effort @ahmedAlaaInstabug !! Please check my suggestions. 🙏

File: android/src/test/java/com/instabug/reactlibrary/RNInstabugCrashReportingModuleTest.java [hunk 1]
```diff
@@ -1,9 +1,85 @@
...
+    public void givenTrue$SetNDKCrashesEnabledShouldCallNativeApiWithEnable() {
+        // when
+        rnModule.setNDKCrashesEnabled(true);
+
+//then
+        mockCrashReporting.verify(() -> CrashReporting.setNDKCrashesState(Feature.State.ENABLED));
+    }
+
+    @Test
+    public void givenTrue$SetNDKCrashesEnabledShouldCallNativeApiWithDisabled() {
```
[the remark this answers is not in the document]
Comment (ahmedAlaaInstabug) [reply android/src/test/java/com/instabug/reactlibrary/RNInstabugCrashReportingModuleTest.java]: @abdelhamid-f-nasser fixed

**Review by a7medev — CHANGES_REQUESTED** [review]
LGTM 👏🏼, please check the comment I added. Let's also add a changelog entry as Abdelhamid suggested 🙏🏼.

File: src/modules/CrashReporting.ts [hunk 2]
```diff
@@ -17,5 +18,15 @@ export const setEnabled = (isEnabled: boolean) => {
  * @param error Error object to be sent to Instabug's servers
  */
 export const reportError = (error: ExtendedError) => {
-  return InstabugUtils.sendCrashReport(error, NativeCrashReporting.sendHandledJSCrash);
+  InstabugUtils.sendCrashReport(error, NativeCrashReporting.sendHandledJSCrash);
```
[the remark this answers is not in the document]
Comment (ahmedAlaaInstabug) [reply src/modules/CrashReporting.ts]: fixed @a7medev

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+
+## [Unreleased](https://github.com/Instabug/Instabug-React-Native/compare/v12.7.0...dev)
+
+### Added
+
+- Add support for enable NDK crashes on Android ([#1132](https://github.com/Instabug/Instabug-React-Native/pull/1132))
```
Comment (abdelhamid-f-nasser) [inline CHANGELOG.md:8]: How about we change it to either one of the below ?
```suggestion
- Add support for enabling NDK crash capturing on Android ([#1132](https://github.com/Instabug/Instabug-React-Native/pull/1132))
```

```suggestion
- Support enabling NDK crash capturing on Android ([#1132](https://github.com/Instabug/Instabug-React-Native/pull/1132))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## [Unreleased](https://github.com/Instabug/Instabug-React-Native/compare/v12.7.1...dev)
+
+### Added
+
+- Support enabling NDK crash capturing on Android ([#1132](https://github.com/Instabug/Instabug-React-Native/pull/1132))
```
Comment (abdelhamid-f-nasser) [inline CHANGELOG.md:7]: Lets add a the missing `.` 🙈 .

```suggestion
- Support enabling NDK crash capturing on Android ([#1132](https://github.com/Instabug/Instabug-React-Native/pull/1132)).
```

**Review by abdelhamid-f-nasser — APPROVED** [review]

**Pull request merged**